fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+      "description" : "Ground Sample Distance",
+      "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/instrument.json#/properties/gsd"
+    },
+    "assets_bands" : {
+      "description" : "Asset eo:bands common names",
+      "$ref": "https://stac-extensions.github.io/eo/v1.0.0/schema.json#/properties/eo:bands/common_name"    
+    }
+  }
+}
+```
+
+Alternately, the client could retrieve the queryables for a single collection with 
+`/collections/collections1/queryables`, which may have more queryables than available for the entire catalog, since
+there may be queryables that are only relevant to that collection. 
+
+Notice in this schema that instead of directly defining the type information about each field, we have instead used the JSON Schema `$ref` mechanism to refer to a STAC schema definition. This not only allows the reuse of these JSON Schema definitions, but also binds an arbitrarily-named Queryable to a specific STAC field. For example, in the above we know that the `eo:cloud_cover` field is referring to the field of the same name in the EO Extension not because they happen to have the same name, but rather because the `$ref` indicates it. The field could just as well be named "cloud_cover", "CloudCover", or "cc", and we would still know it filtered on the EO extension `eo:cloud_cover` field.
+
+While these do seem quite complex to write and understand, keep in mind that query construction will likely be done with a more ergonomic SDK, and query parsing will be done with the help of a ABNF grammar and OpenAPI schema.
+
+From the Queryables above, a client could then form the following example filter expressions. 
+
+These parameters/fields must be supported by the Item Search endpoint and Items resource (`/collections/$collectionId/items`).
+
+## Examples
+
+Note: the GET examples with query parameters are unescaped to make them easier to read. 
+
+### Example 1
+
+This example uses the queryables definition in (Interaction with Endpoints)(#interaction-with-endpoints).
+
+#### Example 1: GET with cql-text
+
+Note that `filter-lang` defaults to `cql-text` in this case, so this is only shown for completeness. 
+
+```http
+GET /search?filter-lang=cql-text&filter=id='LC08_L1TP_060247_20180905_20180912_01_T1_L1TP' AND collection='landsat8_l1tp'
+```
+
+#### Example 1: POST with cql-json
+
+Note that `filter-lang` defaults to `cql-json` and `filter-crs` defaults to `http://www.opengis.net/def/crs/OGC/1.3/CRS84` in this case.
+
+```http
+POST /search
+{ 
+  "filter-lang"="cql-json",
+  "filter-crs":"http://www.opengis.net/def/crs/OGC/1.3/CRS84",
+  "filter": {
+    "and": [
+      "eq": [
+        { "property": "id" },
+        "LC08_L1TP_060247_20180905_20180912_01_T1_L1TP"
+      ],
+      "eq": [
+        { "property": "collection" },
+        "landsat8_l1tp"
+      ]
+    ]
+  }
+}
+```
+
+### Example 2
+
+This example uses the queryables definition in (Interaction with Endpoints)(#interaction-with-endpoints).
```

Comment:
Should perhaps note that the collection field only really makes sense for item-search queries, not for ogcapi-features queries, as those are already against a single collection.